### PR TITLE
[Synfig Studio] Stop rendering button has been moved to the right of the progress bar

### LINKED ISSUE #2037: Improve render progress info widgets
**Synfig version & platform**:
Building from source
Fedora 33

**Issue description**:
Screenshot from 2021-02-19 16-04-37

Show that button only when an animation project is being exported and then hide it again after it's done.

UPDATE: The intention of this issue changed based on discussions below.

### PR DESCRIPTION
**Pull request by FirasH**
Fix #2037

Before:
Screenshot_106a

After:
Screenshot_105a

### CODE REVIEW

File: synfig-studio/src/gui/docks/dock_info.cpp [hunk 1]
```diff
@@ -146,45 +146,52 @@ studio::Dock_Info::Dock_Info()
...
-	table->attach(*separator2, 0,4,8,1);
+	grid->attach(*separator2, 0, 4, 8, 1);
 
-	//Render Progress Bar
+	// Render Progress Bar
+	Gtk::HBox *render_box = manage(new Gtk::HBox());
 	Gtk::Label *render_progress_label = manage(new Gtk::Label());
-	render_progress_label->set_markup(etl::strprintf("<b>%s</b>", _("Render Progress: ")));
-	table->attach_next_to(*render_progress_label, *separator2, Gtk::POS_BOTTOM, 8,1);
+	Gtk::Label *left_spacer_label = manage(new Gtk::Label());
```
Comment (Keyikedalube) [inline synfig-studio/src/gui/docks/dock_info.cpp:160]: So this label caused the start margin to be bigger than the end margin

screenshot

Comment (FirasH) [reply synfig-studio/src/gui/docks/dock_info.cpp:160]: I've added `left_spacer_label` as a space on the left side of the progress bar in order to center it vertically as shown here:
|<--left_space-->|progress-bar|<--right_space-->|
but maybe there's a better method? Didn't find any horizontal spacers or space holders in gtkmm.

File: synfig-studio/src/gui/docks/dock_info.cpp [hunk 2]
```diff
@@ -146,45 +146,52 @@ studio::Dock_Info::Dock_Info()
...
 	render_progress.set_hexpand(true);
 	render_progress.set_vexpand(false);
 	render_progress.set_show_text(true);
 	render_progress.set_text(strprintf("%.1f%%", 0.0));
 	render_progress.set_fraction(0.0);
 
-	stop_button.set_label(_("Stop rendering"));
+	stop_button.set_image_from_icon_name("process-stop", Gtk::IconSize::from_name("synfig-small_icon"));
+	stop_button.set_hexpand(false);
+	stop_button.set_vexpand(false);
```
[the remark this answers is not in the document]
Comment (FirasH) [reply synfig-studio/src/gui/docks/dock_info.cpp:174]: Yes, you're right. I'll remove them 👍

File: synfig-studio/src/gui/docks/dock_info.cpp (hunk 1 above)
Comment (Keyikedalube) [reply synfig-studio/src/gui/docks/dock_info.cpp:160]: `set_margin_start(30)`?

Comment (Keyikedalube) [reply synfig-studio/src/gui/docks/dock_info.cpp:160]: And same for stop button left margin that way widgets look properly center aligned.

File: synfig-studio/src/gui/docks/dock_info.cpp [hunk 3]
```diff
@@ -157,30 +157,30 @@ studio::Dock_Info::Dock_Info()
...
-	render_progress.set_show_text(true);
-	render_progress.set_text(strprintf("%.1f%%", 0.0));
+	render_progress.set_margin_end(5);
+	render_progress.set_valign(Gtk::ALIGN_CENTER);
 	render_progress.set_fraction(0.0);
 
+	render_percentage.set_text(strprintf("%.1f%%", 0.0));
+
+	overlay->add_overlay(render_progress);
+	overlay->add_overlay(render_percentage);
```
Comment (Keyikedalube) [inline synfig-studio/src/gui/docks/dock_info.cpp:175]: Shouldn't `render_progress` be added to a `Gtk::Container`? According to [GTKMM 3.24 docs](https://developer.gnome.org/gtkmm/3.24/classGtk_1_1Overlay.html#a6e8eb1201093c8b9c0ea7f0f274ae587) on `add_overlay()`:

> The widget will be stacked on top of the main widget added with **Gtk::Container::add()**.

@ice0's example in this https://github.com/synfig/synfig/issues/2037#issuecomment-808842732 also adds the progress to a container and overlays the label on top of that progress bar
```python
...
    ...
        overlay.add(self.progressbar)

        self.plabel = Gtk.Label.new()
        overlay.add_overlay(self.plabel)
    ...
...
```

GTK YouTuber also does the same [Tutorial 3 at timestamp 3:35](https://youtu.be/Jj0sXiJ0ENU?t=335)

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**